### PR TITLE
Really build quic_platform_impl_lib as a library.

### PR DESCRIPTION
**Pull request by wu-bin**
*Description*:

Really build quic_platform_impl_lib as a library. Currently this target only has "hdrs", but no "srcs", so "bazel build :quic_platform_impl_lib" is a noop which succeeds even if the impl header files are not compilable.

*Risk Level*: minimal, build only
*Testing*: 

1. bazel build :quic_platform_impl_lib  : works
2. bazel build :quic_platform_impl_lib, with a typo in a quic impl header file  :  does not build
3. bazel test test/extensions/quic_listeners/quiche/platform:quic_platform_test --test_output=all:  All tests pass

*Docs Changes*: none
*Release Notes*: none
[Optional Fixes #Issue]
[Optional *Deprecated*:]

### CODE REVIEW

File: source/extensions/quic_listeners/quiche/platform/BUILD [hunk 1]
```diff
@@ -37,23 +37,40 @@ envoy_cc_library(
...
+    "quic_iovec_impl.h",
+    "quic_prefetch_impl.h",
+    "quic_ptr_util_impl.h",
+    "quic_string_impl.h",
+    "quic_string_piece_impl.h",
+    "quic_uint128_impl.h",
+]
+
+genrule(
+    name = "generate_dummy_src_for_quic_header_only_impl",
```
Comment (mpwarres) [inline source/extensions/quic_listeners/quiche/platform/BUILD]: shorten to "generate_quic_header_only_impl"?

File: source/extensions/quic_listeners/quiche/platform/BUILD [hunk 2]
```diff
@@ -37,23 +37,40 @@ envoy_cc_library(
...
+    "quic_flag_utils_impl.h",
+    "quic_iovec_impl.h",
+    "quic_prefetch_impl.h",
+    "quic_ptr_util_impl.h",
+    "quic_string_impl.h",
+    "quic_string_piece_impl.h",
+    "quic_uint128_impl.h",
+]
+
+genrule(
```
Comment (mpwarres) [inline source/extensions/quic_listeners/quiche/platform/BUILD]: Add brief comment explaining the reason for this. E.g:

```
# Generate dummy .cc file that includes all platform headers, to ensure that they
# will actually be compiled.
#
# TODO(wub): remove once real dependencies from core QUICHE code to platform
#   libraries are added.
```

File: source/extensions/quic_listeners/quiche/platform/BUILD [hunk 3]
```diff
@@ -37,23 +37,40 @@ envoy_cc_library(
...
-        "quic_export_impl.h",
-        "quic_fallthrough_impl.h",
-        "quic_flag_utils_impl.h",
-        "quic_iovec_impl.h",
-        "quic_prefetch_impl.h",
-        "quic_ptr_util_impl.h",
-        "quic_string_impl.h",
-        "quic_string_piece_impl.h",
-        "quic_uint128_impl.h",
+    srcs = [
```
Comment (mpwarres) [inline source/extensions/quic_listeners/quiche/platform/BUILD]: Can you add this for {http2,spdy}_platform_impl_lib as well?

Comment (wu-bin) [reply source/extensions/quic_listeners/quiche/platform/BUILD]: Done.

File: source/extensions/quic_listeners/quiche/platform/BUILD (hunk 2 above)
Comment (wu-bin) [reply source/extensions/quic_listeners/quiche/platform/BUILD]: Done.

File: source/extensions/quic_listeners/quiche/platform/BUILD (hunk 1 above)
Comment (wu-bin) [reply source/extensions/quic_listeners/quiche/platform/BUILD]: Done.

**Pull request closed without merging**